### PR TITLE
[WIP] poc: use SFC for Vue components

### PR DESCRIPTION
**Pull request by carwack**
This is a PoC to see if we can use Vue SFC for the Ark components.
This solves TS issues or missing TS information.

I believe the code can use some improvement, but it's a working idea on the Avatar component.

### CODE REVIEW

File: packages/frameworks/vue/src/avatar/Avatar.vue [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<script setup lang="ts">
+import type { VNode } from 'vue'
+import { ark, type HTMLArkProps } from '../factory'
+import type { Assign } from '../types'
+import { AvatarProvider } from './avatar-context'
+import { props as avProps } from './avatar.props' // Name change was needed to not have double props
```
Comment (Omikorin) [inline packages/frameworks/vue/src/avatar/Avatar.vue:6]: We can edit a vue props script to generate the object named `componentProps` instead of props.

**Comment by iamdin** [comment]
I took a look at the usage of other component libraries:
- Radix Vue use SFC and `<script setup>`;
- NuxtUI use SFC;
- HeadlessUI, Vuetify and Antdv use TSX;

File: packages/frameworks/vue/src/avatar/Avatar.vue (hunk 1 above)
Comment (carwack) [reply packages/frameworks/vue/src/avatar/Avatar.vue]: I made a change into renaming the props in `avatar.props.ts` to `componentProps`.

**Comment by carwack** [comment]
I did learn a bit more about the `attrs` and when we don't want to inherit them or not. This vid helped me out: https://www.youtube.com/watch?v=tcD5TbOYHOw&ab_channel=LearnVue

File: packages/frameworks/vue/src/avatar/avatar.props.ts [hunk 2]
```diff
@@ -1,16 +1,23 @@
-import type { Context } from '@zag-js/avatar'
+import type { Context, StatusChangeDetails } from '@zag-js/avatar'
 import type { PropType } from 'vue'
-import { declareEmits } from '../utils'
 
-export const props = {
+export const avatarProps = {
+  /** The document's text/writing direction. */
   dir: {
+    default: 'ltr',
```
Comment (Omikorin) [inline packages/frameworks/vue/src/avatar/avatar.props.ts:7]: As far as I know, dir defaults to user's browser preference, so we shouldn't set this

Comment (carwack) [reply packages/frameworks/vue/src/avatar/avatar.props.ts]: Reverted this, good point!

**Comment by segunadebayo** [comment]
Closing this out for now, to avoid stale PRs

Kindly re-open a PR when you have a complete PR ready for review.

Appreciate the efforts 👍

**Pull request closed without merging**